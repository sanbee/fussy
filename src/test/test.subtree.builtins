/* -*- C -*- */
line="------------------------------------------------------\n";
msg="Error on the above expression should be = ";

test()
{
  auto x,y,x1,x2,tt,s,tsq,ssq,c,as,ac,at,sh,ch,th,ash,ach,l,e,em,s2,c2,t,ath;
  print "----------------------------------------------------------------------------\n";
  print "Script to test error propogation using builtin functions and sub-expressions\n"
  print "----------------------------------------------------------------------------\n";
  x=1pm0.1;
  y=2pm0.2;
  x1=1pm0.1;
  x2=1pm0.1;
  print "Value of x=\t",x,"\n";
  print "Value of y=\t",y,"\n";
  print "Value of x1=\t",x1,"\n";
  print "Value of x2=\t",x2,"\n";
  print line;
    
  s:=sin(x);
  print "s:=sin(x)\n";
  print "sin(x)=\t",sin(x),"\n";
  print "s=\t",s,"\n";
  print "sqrt(1-s^2)=\t",sqrt(1-s^2),"\n";
  ssq:=s^2;
  print "ssq:=s^2\n";
  print "sqrt(1-ssq)=\t",sqrt(1-ssq),"\n";
  print line;
  
  tt:=s;
  tsq:=tt^2;
  print "tt:=s\n";
  print "tsq:=tt^2\n";
  print "s= ",s,"\nssq= ",ssq,"\ntt= ",tt,"\ntsq= ",tsq,"\n";
  print line;
  
  c:=cos(x);
  print "c:=cos(x)\n";
  print "cos(x)=\t",cos(x),"\n";
  print line;
  
  t:=tan(x);
  print "t:=tan(x)\n";
  print "tan(x)=\t",tan(x),"\n";
  print "t=\t",t,"\n";
  print line;
  
  print "sin(x)/cos(x)=\t",sin(x)/cos(x),"\n";
  print "s/c=\t",s/c,"\n";
  print line;
  
  as:=asin(s);
  print "as:=asin(s);\n";
  print "asin(sin(x))=\t",asin(sin(x)),"\n";
  print "asin(s) as a partial var.=\t",as,"\n";
  print line;
  
  ac:=acos(c);
  print "ac:=acos(c);\n";
  print "acos(cos(x))=\t",acos(cos(x)),"\n";
  print "acos(c) as a partial var.=\t",ac,"\n";
  print line;
  
  at:=atan(t);
  print "at:=atan(t);\n";
  print "atan(tan(x))=\t",atan(tan(x)),"\n";
  print "atan(t) as a partial var.=\t",at,"\n";
  print line;
  
  sh:=sinh(x);
  ch:=cosh(x);
  th:=tanh(x);
  ash:=asinh(sh);
  ach:=acosh(ch);
  ath:=atanh(th);
  print "sh:=sinh(x)\n";
  print "ch:=cosh(x)\n";
  print "th:=tanh(x)\n";
  print "ash:=asinh(sh)\n";
  print "ach:=acosh(ch)\n";
  print "ath:=atanh(th)\n";
  
  print "asinh(sinh(x))=\t",asinh(sinh(x)),"\n";
  print "ash=\t",ash,"\n";
  print line;
  
  print "acosh(cosh(x))=\t",acosh(cosh(x)),"\n";
  print "ach=\t",ach,"\n";
  print line;
  
  print "atanh(tanh(x))=\t",atanh(tanh(x)),"\n";
  print "ath=\t",ath,"\n";
  print line;
  
  l:=ln(x);
  print "l:=ln(x)\n";
  print "exp(ln(x))=\t",exp(ln(x)),"\n";
  print "exp(l)=\t",exp(l),"\n";
  print line;
  
  print "sinh(x)=\t",sinh(x),"\n";
  print "sh=\t",sh,"\n";
  print "(exp(x)-exp(-x))/2=\t",(exp(x)-exp(-x))/2,"\n";
  e:=exp(x);em:=exp(-x);
  print "e:=exp(x);em:=exp(-x)\n";
  
  print "e=\t",e,"\n";
  print "em=\t",em,"\n";
  print "(e-em)/2=\t",(e-em)/2,"\n";
  print line;
  
  print "sinh(x)/((exp(x)-exp(-x))/2)=\t",sinh(x)/((exp(x)-exp(-x))/2),"\n";
  print "sh/((e-em)/2)=\t",sh/((e-em)/2),"\n";
  print line;
  
  print "x/exp(ln(x))=",x/exp(ln(x)),"\n";
  print "x/exp(l)=",x/exp(l),"\n";
  print line;
  
  print "y/(1-y) =",(y/(1-y))%16.6f,"\n";
  print "y/(-y+1)=",(y/(-y+1))%16.6f,"\n";
  print line;
  
  s2:=sin(x2);
  c2:=cos(x2);
  print "s2:=sin(x2)\n";
  print "c2:=cos(x2)\n";
  
  print "sin(x1)*sin(x1)=\t",sin(x1)*sin(x1),"\n";
  print "s*s=\t",s*s,"\n";
  print line;
  
  print "sin(x1)*sin(x2)=\t",sin(x1)*sin(x2),"\n";
  print "s*s2=\t",s*s2,"\n";
  print line;
  
  print "(sin(x1)^2+cos(x1)^2)=\t",(sin(x1)^2+cos(x1)^2),"\n";
  print "(s^2+c^2)=\t",(s^2+c^2),"\n";
  print line;
  
  print "(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2)=\t",(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2),"\n";
  print "(s^2+c^2)/(s2^2+c2^2)=\t",(s^2+c^2)/(s2^2+c2^2),"\n";
  print line;
  
  print "sin(x1)^2+cos(x2)^2=\t",sin(x1)^2+cos(x2)^2,"\n";
  print "s^2+c2^2=\t",s^2+c2^2,"\n";
  print line;
  
  x=2pm0.2;
  y=3pm0.3;
  print "sin(x)^cos(x) = ",sin(x)^cos(x),"\n";
  print msg,.2*(sin(x)^cos(x))* (cos(x)*cos(x)/sin(x) - log(sin(x))*sin(x)),"\n";
  
  print "x^y = ",x^y,"\n";
  print msg,(x^y)*sqrt(((y/x)*0.2)^2+(log(x)*0.3)^2),"\n";
  
  print line;
}

test();
