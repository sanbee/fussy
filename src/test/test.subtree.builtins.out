----------------------------------------------------------------------------
Script to test error propogation using builtin functions and sub-expressions
----------------------------------------------------------------------------
Value of x=	   1.00000 +/-    0.10000
Value of y=	   2.00000 +/-    0.20000
Value of x1=	   1.00000 +/-    0.10000
Value of x2=	   1.00000 +/-    0.10000
------------------------------------------------------
s:=sin(x)
sin(x)=	   0.84147 +/-    0.05403
s=	   0.84147 +/-    0.05403
sqrt(1-s^2)=	   0.54030 +/-    0.08415
ssq:=s^2
sqrt(1-ssq)=	   0.54030 +/-    0.08415
------------------------------------------------------
tt:=s
tsq:=tt^2
s=    0.84147 +/-    0.05403
ssq=    0.70807 +/-    0.09093
tt=    0.84147 +/-    0.05403
tsq=    0.70807 +/-    0.09093
------------------------------------------------------
c:=cos(x)
cos(x)=	   0.54030 +/-    0.08415
------------------------------------------------------
t:=tan(x)
tan(x)=	   1.55741 +/-    0.34255
t=	   1.55741 +/-    0.34255
------------------------------------------------------
sin(x)/cos(x)=	   1.55741 +/-    0.34255
s/c=	   1.55741 +/-    0.34255
------------------------------------------------------
as:=asin(s);
asin(sin(x))=	   1.00000 +/-    0.10000
asin(s) as a partial var.=	   1.00000 +/-    0.10000
------------------------------------------------------
ac:=acos(c);
acos(cos(x))=	   1.00000 +/-    0.10000
acos(c) as a partial var.=	   1.00000 +/-    0.10000
------------------------------------------------------
at:=atan(t);
atan(tan(x))=	   1.00000 +/-    0.10000
atan(t) as a partial var.=	   1.00000 +/-    0.10000
------------------------------------------------------
sh:=sinh(x)
ch:=cosh(x)
th:=tanh(x)
ash:=asinh(sh)
ach:=acosh(ch)
ath:=atanh(th)
asinh(sinh(x))=	   1.00000 +/-    0.10000
ash=	   1.00000 +/-    0.10000
------------------------------------------------------
acosh(cosh(x))=	   1.00000 +/-    0.10000
ach=	   1.00000 +/-    0.10000
------------------------------------------------------
atanh(tanh(x))=	   1.00000 +/-    0.10000
ath=	   1.00000 +/-    0.10000
------------------------------------------------------
l:=ln(x)
exp(ln(x))=	   1.00000 +/-    0.10000
exp(l)=	   1.00000 +/-    0.10000
------------------------------------------------------
sinh(x)=	   1.17520 +/-    0.15431
sh=	   1.17520 +/-    0.15431
(exp(x)-exp(-x))/2=	   1.17520 +/-    0.15431
e:=exp(x);em:=exp(-x)
e=	   2.71828 +/-    0.27183
em=	   0.36788 +/-    0.03679
(e-em)/2=	   1.17520 +/-    0.15431
------------------------------------------------------
sinh(x)/((exp(x)-exp(-x))/2)=	   1.00000
sh/((e-em)/2)=	   1.00000
------------------------------------------------------
x/exp(ln(x))=   1.00000
x/exp(l)=   1.00000
------------------------------------------------------
y/(1-y) =       -2.000000 +/-         0.200000
y/(-y+1)=       -2.000000 +/-         0.200000
------------------------------------------------------
s2:=sin(x2)
c2:=cos(x2)
sin(x1)*sin(x1)=	   0.70807 +/-    0.09093
s*s=	   0.70807 +/-    0.09093
------------------------------------------------------
sin(x1)*sin(x2)=	   0.70807 +/-    0.06430
s*s2=	   0.70807 +/-    0.06430
------------------------------------------------------
(sin(x1)^2+cos(x1)^2)=	   1.00000
(s^2+c^2)=	   1.00000
------------------------------------------------------
(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2)=	   1.00000
(s^2+c^2)/(s2^2+c2^2)=	   1.00000
------------------------------------------------------
sin(x1)^2+cos(x2)^2=	   1.00000 +/-    0.12859
s^2+c2^2=	   1.00000 +/-    0.12859
------------------------------------------------------
sin(x)^cos(x) =    1.04036 +/-    0.05762
Error on the above expression should be =    0.05762 +/-    0.05712
x^y =    8.00000 +/-    2.92017
Error on the above expression should be =    2.92017 +/-    1.14554
------------------------------------------------------
